ballerina-shell/modules/shell-cli/src/main/java/io/ballerina/shell/cli/BallerinaShell.java

```diff
@@ -235,4 +265,81 @@ public void reset() {
+        List<String> missingModules = new ArrayList<>();
+        List<String> requiredModules = new ArrayList<>();
+        terminal.info("Found following undefined module(s).");
+        for (String module : modules) {
+            module = module.replaceAll("'", "");
+            if (moduleImporter.isModuleInDistRepo(module)) {
+                requiredModules.add(module);
+            } else {
+                missingModules.add(module);
+            }
+            terminal.info(module);
+        }
+
+        if (requiredModules.size() > 0) {
+            terminal.info("\nFollowing undefined modules can be imported.");
+            int moduleCount = 1;
+            for (String module : requiredModules) {
+                terminal.info(String.format("%d. %s", moduleCount, module));
+                moduleCount += 1;
+            }
+            
+            String answer = terminal.readOneLine("Do you want to import mentioned modules (yes/y) (no/n)? ");
+            if (answer.equalsIgnoreCase("yes") || answer.equalsIgnoreCase("y")) {
+                for (String module : requiredModules) {
+                    String importSource = moduleImporter.getImportStatement(module);
+                    terminal.info("\nAdding import: " + importSource);
+                    try {
+                        ShellCompilation shellCompilation = evaluator.getCompilation(importSource);
+                        Optional<PackageCompilation> compilation = shellCompilation.getPackageCompilation();
+                        evaluator.getValue(compilation);
+                        terminal.info("Import added: " + importSource);
+                    } catch (BallerinaShellException ex) {
+                        terminal.error("Error occurred while adding imports.");
+                        outputException(ex);
+                    }
+                }
+
+                if (missingModules.size() > 0) {
+                    terminal.error("\nFound following missing module(s).");
+                    for (String missingModule : missingModules) {
+                        terminal.error(missingModule);
+                    }
+                }
+            } else {
+                terminal.error("\nFound missing module(s).");
+            }
+
+        } else {
+            terminal.error("\nFound missing module(s).");
+        }
+    }
+
+    /**
+     * Return found undefined modules.
+     *
+     * @param diagnostics collection of evaluator diagnostics.
+     * @return distinct list of module errors.
+     */
+    public boolean isContainsUndefinedModules(Collection<Diagnostic> diagnostics) {
+        boolean isContainsUndefinedModules = false;
+        for (Diagnostic diagnostic : diagnostics) {
+            if (diagnostic.toString().contains("undefined module")) {
+                isContainsUndefinedModules = true;
+                break;
+            }
+        }
+
+        return isContainsUndefinedModules;
```

Comment:
Let's 
```suggestion
        for (Diagnostic diagnostic : diagnostics) {
            if (diagnostic.toString().contains("undefined module")) {
                return true;
            }
        }

        return false;
```

Reply:
Updated in 7206767a98cd266b722f016fca60ecf1fac9aec7